fim: fix cross / strictDeps build

FliegendeWurst opened the pull request:
ref. #178468

## Things done

- Built on platform(s)
  - [x] x86_64-linux: and cross to aarch64
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

FliegendeWurst committed:
fim: fix cross / strictDeps build

uninsane commented on pkgs/by-name/fi/fim/package.nix:
```diff
@@ -35,22 +36,31 @@ stdenv.mkDerivation rec {
 
+  patches = [
+    # build tools with a build compiler
+    ./native-tools.patch
```
the better way to do this is to use `@cc_for_build@` in the .patch file and then substitute here instead of in `postPatch`:
```nix
patches = [
  (substituteAll {
    src = ./native-tools.patch;
    cc_for_build = lib.getExe buildPackages.stdenv.cc;
  })
];
```

see pkgs/by-name/gn/gnome-shell/package.nix for an example

uninsane reviewed (dismissed):
haven't tried building it, but aside from the patch substitution the rest LGTM

uninsane commented:
new version breaks native compilation. error message suggests to use `substitute` (or `substituteAll`)

```
$ nix-build -A fim
these 2 derivations will be built:
  /nix/store/abf4yr0jwmginynh2mk7vip8q5czq9y0-native-tools.patch.drv
  /nix/store/w2q0r4s43xlzcc0a6bsl1s9xs5sc3wks-fim-0.7.drv
building '/nix/store/abf4yr0jwmginynh2mk7vip8q5czq9y0-native-tools.patch.drv'...
Running phase: patchPhase
Running phase: updateAutotoolsGnuConfigScriptsPhase
Running phase: configurePhase
no configure script, doing nothing
Running phase: buildPhase
Running phase: checkPhase
The following look like unsubstituted Nix identifiers that remain in /nix/store/118rabdw0xy671p8ijsdxnxvn1g7nbxb-native-tools.patch:
@FIM_WANT_CUSTOM_HARDCODED_CONSOLEFONT_TRUE@
@HAVE_RUNNABLE_TESTS_TRUE@
@FIM_WANT_CUSTOM_HARDCODED_CONSOLEFONT_TRUE@
@HAVE_RUNNABLE_TESTS_TRUE@
Use the more precise `substitute` function if this check is in error.
```

FliegendeWurst commented:
Fixed. Bit of a silly check since uppercase variables can't be substituted anyway #28086

uninsane commented:
```
$ nix-build -A pkgsCross.aarch64-multiplatform.fim
...
checking for libpng-config... no
```
native build has png support: would ideally also have png support when cross compiling (or at the very least, change up the `pngSupport` package option to reflect reality).

FliegendeWurst commented:
Since we switched SDL to SDL_compat -> sdl2-compat -> sdl3 -> ... -> Qt, x11Support now defaults to false for cross.

libpng fixed.

uninsane commented:
```
$ nix-build -A pkgsCross.aarch64-multiplatform.fim
$ head -n1 ./result/bin/fimgs
#!/bin/bash
```

needs its shebang fixed

uninsane commented:
also, cross is building without ascii art support:
```
configure: Enabling aalib driver.
checking for aalib-config... no
configure: WARNING: ***  no aalib-config ? ***
```
can likely be fixed the same way png was fixed.

other than that i think it's correct.

FliegendeWurst commented:
Fixed, and I've now just compared the build logs, as I should have perhaps done from the start. And found two more things that differed, fixed via configureFlags.
